### ISSUE TITLE
unmarshal errors field ruler not found in type main.Config

### ISSUE THREAD

**Issue by 10101010101010001** [grafana/loki#2706]
Hi,

I am unable to run the latest release of loki, I've followed the install instructions using the binaries. 

Details:
`./loki-linux-amd64 -config.file loki-local-config.yaml
failed parsing config: loki-local-config.yaml: yaml: unmarshal errors:
  line 47: field ruler not found in type main.Config` 

Config:
https://raw.githubusercontent.com/grafana/loki/master/cmd/loki/loki-local-config.yaml

System: 
5.4.0-42-generic

Thanks.

**Comment by zen** [comment]
I'm experiencing exactly the same issue using provided example configs. Digging deeper

**Comment by zen** [comment]
Looks like Loki is not aware of the new `ruler` config top entry and throwing error on validation.

Not an Go expert, but ruler was introduced in #1939  and I don't see anything on extending config validation in that PR.

**Comment by zen** [comment]
@owen-d you might want to take a look and confirm / call me idiot ;)

**Comment by owen-d** [comment]
Hey sorry about this one. It's because we rolled out the docs that reference the master branch (our site only supports `latest` right now). We're prepping a `1.7` release in the next few days that will include this, but right now it's only in the master branch.

**Comment by zen** [comment]
What about default configs? Same story?

**Comment by owen-d** [comment]
I realize I wasn't very clear in my earlier comment. The `ruler` field doesn't exist in the 1.6 release, so trying to include it in the yaml config will throw an error because Loki errors upon finding an unrecognized field. If you use a build from the master branch the `ruler` config should be parsed correctly or alternatively you can wait for the 1.7 release (a few days away). Does that help?

> What about default configs? Same story?
Can you rephrase? I'm not sure what the question is.

**Comment by zen** [comment]
That's actually me not being clear. I understand that master branch is actually ahead, including sample configs

**Comment by 10101010101010001** [comment]
Removing the ruler element fixed the issue, thanks guys.

**Comment by owen-d** [comment]
Great, closing this.
